components/fxa-client/src/send_tab.rs

```diff
@@ -38,16 +40,21 @@ impl FirefoxAccount {
     }
 
     /// Send a single tab to another device designated by its device ID.
+    /// XXX - this should change to taking a Vec of device IDs, because
+    /// collecting telemetry relies on a multi-device send happening in a single
+    /// operation.
     pub fn send_tab(&mut self, target_device_id: &str, title: &str, url: &str) -> Result<()> {
```

Comment:
I wonder, how do you feel about this method returning a `Result<SentReceivedTab>`—maybe rename it `SendTabTelemetry`—instead of indirecting through `FxaTelemetry`? I was thinking it would be easier for the caller to get the IDs back right away, which it can then either stash in the Sync ping (iOS) or Glean (Fenix), instead of sending the tab, then turning around and calling `fxa_gather_telemetry` right away.

Reply:
IOW, I guess I'm wondering, why shouldn't we pollute the API with telemetry? 😄 A return value means the caller has to handle it, but collecting it separately means the caller has to turn around and call `gather_telemetry` immediately after sending the tab or handling an incoming "tab received" command—and forgetting to do that means we drop the telemetry on the floor.